src/main/java/net/sf/jabref/gui/actions/ManageKeywordsAction.java

```diff
@@ -313,7 +314,9 @@ private NamedCompound updateKeywords(List<BibEntry> entries, Set<String> keyword
             }
 
             if (Globals.prefs.isKeywordSyncEnabled()) {
-                SpecialFieldsUtils.syncSpecialFieldsFromKeywords(entry, ce);
+                List<FieldChange> changes = new ArrayList<>();
```

Comment:
Could you maybe do a little refactoring here? The `changes` list is really a return argument. So instead of having the caller of `syncSpecialFieldsFromKeywords` create the list and pass it to the method, it would be better if the method created the list itself and just returns it to the caller. Return arguments are common in C, but considered really bad style in Java. 

This change would propagate to the other `synchSpecialFields` methods and require some rewriting. If I did not miss anything, there is are only internal calls in `SpecialFieldsUtil` that might have a list that has actual content inserted into these methods. Even in this case, it would be better to create a new list as return object internally and copy the contents of the list inserted as argument is needed.

This is really not the fault of this PR, but the bad design of special fields which should be fixed.